src/components/EkoSubtitles/EkoSubtitles.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {useEffect, useState} from 'react';
+import "./EkoSubtitles.scss";
+// https://meta.wikimedia.org/wiki/Template:List_of_language_names_ordered_by_code
+const RTL_LANGUAGES = ['ar', 'arc', 'dv', 'fa', 'ha', 'he', 'khw', 'ks', 'ku', 'ps', 'ur', 'yi'];
+const AD_LANGUAGES = ['en-US-AD'];
+
+export function EkoSubtitles({player, style}) {
+    // TODO: Do I need to useRef() or useState() for the player?
+
+    const [visible, setVisible] = useState(false);
+    const [text, setText] = useState('');
+    const [effectiveLang, setEffectiveLang] = useState('');
+
+    useEffect(() => {
+        if (!player) {
+            return;
+        }
+        const onSubtitlesInit = () => player.invoke('subtitles.mode', 'proxy');
+        const onVisibilityChange = (isVisible) =>  setVisible(isVisible);
+        const onSubStart = (subObj) =>  setText(subObj.text);
+        const onSubEnd = (subObj) =>  setText('');
+        const onLangChange = (effectiveLanguage) =>  setEffectiveLang(effectiveLanguage);
+
+        player.on('plugininitsubtitles', onSubtitlesInit);
+        player.on('subtitles.visibilitychange', onVisibilityChange);
+        player.on('subtitles.substart', onSubStart);
+        player.on('subtitles.subend', onSubEnd);
+        player.on('subtitles.effectivelanguagechange', onLangChange);
+        
+        return () => {
+            player.off('plugininitsubtitles', onSubtitlesInit);
+            player.off('subtitles.visibilitychange', onVisibilityChange);
+            player.off('subtitles.substart', onSubStart);
+            player.off('subtitles.subend', onSubEnd);
+            player.off('subtitles.effectivelanguagechange', onLangChange);
+        };
+    }, [player]);
+
+    // Add right-to-left "direction" css for required languages.
+    let rtl = RTL_LANGUAGES.includes(effectiveLang) ? 'rtl' : '';
+    let isAudioDescriptionLanguage = AD_LANGUAGES.includes(effectiveLang);
+    // Aria Live should only be present for Audio Description languages
+    let ariaLiveValue = isAudioDescriptionLanguage ? 'polite' : null;
+
+    return (
+        <div role= "Subtitles" style={style}>
+            {visible && text && <div className={`eko_subtitles_div ${rtl}`} aria-live={ariaLiveValue} >
+                <span>{text}</span>
+            </div>}
+        </div>
+    );
+}
```

Comment:
Requires propTypes and documentation
see `EkoVideo.jsx`

Reply:
@Melatonin64 suggested holding off on fully documenting and **not** making this publicly acknowledged... thoughts?

Reply:
@Melatonin64 Any particular reason? In any case the subtitles component story _will_ be public, so having it public without docs just felt weird to me.

Reply:
I mentioned briefly to Hofshy how the developer experience would be (i.e. grabbing a ref to the `EkoPlayer` instance and passing it into the subtitles component) and he thought we may want to look at changing the API at some point, so it was suggested that we hold off on documenting for now until we are more confident that this is the API.

I'm not sure if there exists a better API, but maybe we can use it for a bit before we commit to it?
I could be convinced otherwise, let me know if you wanna discuss!

Reply:
@Melatonin64 
I think that we can always change the interface if we want to and push out a new major version, and nothing will break.

Having the official documentation to contain the story for the component and be empty is just a bad look for us.
I feel that since it's a big question mark when and this will happen, then this will essentially become a "temporary permanent solution" so I'd prefer to have at least something, rather than nothing at all even if we do change the API later on.

How about just doing minimal docs so that the public docs/story aren't empty?

Reply:
Ok, let's document!
Side note - I wonder if we could improve the API by using a React context?